src/Components/Config/ConfigScreen.js

```diff
@@ -45,8 +60,15 @@ class ConfigScreen extends PureComponent {
     return _.isEqual(this.state.settings, this.props.settings) == false
   }
 
-  handleCancelPressed () {
-    hashHistory.pop();
+  handleCancelPressed = (e) => {
+    if (this.state.cancelIsRestart) {
```

Comment:
based on the usage here I'd suggest a name like `isStartupFailure` or something instead of `cancelIsRestart`